knimin/tests/test_ag_get_participant_names.py

```diff
@@ -0,0 +1,47 @@
+from unittest import main
+
+from tornado.escape import url_escape
+
+from knimin.tests.tornado_test_base import TestHandlerBase
+
+
+class TestAGNamesHandler(TestHandlerBase):
+    def test_get_not_authed(self):
+        response = self.get('/ag_participant_names/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/login/?next=%s' %
+                         (port, url_escape('/ag_participant_names/')))
+
+    def test_get(self):
+        # test if side shows up when properly logged in
+        self.mock_login_admin()
+        response = self.get('/ag_participant_names/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/ag_participant_names/' % port)
+
+    def test_post(self):
+        self.mock_login_admin()
+        response = self.post('/ag_participant_names/', {})
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/ag_participant_names/' % port)
+
+
+class TestAGNamesDLHandler(TestHandlerBase):
+    def test_post(self):
+        self.mock_login_admin()
+        response = self.post('/ag_participant_names/download/', {})
+        self.assertEqual(response.code, 200)
+        # is it sufficient to check that the response provides a zip-file or
+        # should I also unpack the content and compare it to a database query?
```

Comment:
actually, that would be useful if easy. however, if the underlying logic is already tested separately then it is redundant